src/components/assets/XcmNativeAssetList.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div>
+    <div class="container">
+      <div class="row">
+        <span class="text--title">{{ $t('assets.xcmAssets') }}</span>
+      </div>
+
+      <div v-for="t in xcmAssets" :key="t.id">
+        <XcmCurrency
+          :token="t"
+          :handle-modal-xcm-transfer="handleModalXcmTransfer"
+          :handle-modal-xcm-bridge="handleModalXcmBridge"
+        />
+      </div>
+    </div>
+
+    <Teleport to="#app--main">
+      <ModalXcmTransfer
+        :is-modal-xcm-transfer="isModalXcmTransfer"
+        :handle-modal-xcm-transfer="handleModalXcmTransfer"
+        :account-data="accountData"
+        :token="token"
+      />
+
+      <ModalXcmBridge
+        :is-modal-xcm-bridge="isModalXcmBridge"
+        :handle-modal-xcm-bridge="handleModalXcmBridge"
+        :account-data="accountData"
+        :token="token"
+      />
+    </Teleport>
+  </div>
+</template>
+<script lang="ts">
+import { ChainAsset, useBalance, useXcmAssets } from 'src/hooks';
+import { computed, defineComponent, ref } from 'vue';
+import ModalXcmTransfer from './modals/ModalXcmTransfer.vue';
+import ModalXcmBridge from './modals/ModalXcmBridge.vue';
+import XcmCurrency from './XcmCurrency.vue';
+import { useStore } from 'src/store';
+
+export default defineComponent({
+  components: { XcmCurrency, ModalXcmBridge, ModalXcmTransfer },
+  setup() {
+    const isModalXcmTransfer = ref<boolean>(false);
+    const isModalXcmBridge = ref<boolean>(false);
+    const token = ref<ChainAsset | null>(null);
+
+    const store = useStore();
+    const selectedAddress = computed(() => store.getters['general/selectedAddress']);
+    const { accountData } = useBalance(selectedAddress);
+
+    const handleModalXcmTransfer = ({ isOpen, currency }: { isOpen: boolean; currency: any }) => {
+      isModalXcmTransfer.value = isOpen;
+      token.value = currency;
+    };
+
+    const handleModalXcmBridge = ({ isOpen, currency }: { isOpen: boolean; currency: any }) => {
+      isModalXcmBridge.value = isOpen;
+      token.value = currency;
+    };
+
+    const { xcmAssets, handleUpdateTokenBalances } = useXcmAssets();
```

Comment:
![168475187-38f78c79-8e83-419d-b275-6616475db651](https://user-images.githubusercontent.com/40356749/168478562-d73197ab-bde7-4812-8623-36d3e8e1700e.png)

I'm seeing other assets outside of KSM (the Foo bar test token). The XCM assets have an id between `2^64 ~ 2^128-1`. Since the asset pallet is permissionless, anyone can register any asset. I recommend that we keep a manual list of assets that are XCM compatible, and hide assets where the balance is 0 and is not supported by XCM.
For example, if the user holds more then one BAR token, they can see the asset listed on the portal with the transfer button (but no XCM button since it's not supported). However, tokens that support XCM should always be displayed so they can bring the asset to the network.

Reply:
Fixed in this commit. 

https://github.com/AstarNetwork/astar-apps/pull/397/commits/7a02bf1b5276c8820cb5c96235fad54994d0f11c

https://github.com/AstarNetwork/astar-apps/pull/397/commits/c1738e5f976f885442d98b0c5503e3ff4d6fe00f

CC: @sirius651 
![image](https://user-images.githubusercontent.com/92044428/168482075-c022f94c-0a7f-4584-bf55-a01ae1225d85.png)

Reply:
@hoonsubin  
1. Just double check that is the xcm button for the bridge visible only on the KSM or DOT Asset?
2. The fetching asset you introduced does not have user asset balance. https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.shiden.astar.network#/assets/balances Is it correct to import it like the information here?